backend/src/apis/UserApi.js

```diff
@@ -142,6 +142,38 @@ router.put("/auth/change-password", apiAsyncWrapper(async (req, res) => {
   res.send(passwordChanged);
 }));
 
+/**
+ * Reset the user password.
+ */
+router.put("/auth/reset-password", apiAsyncWrapper(async (req, res) => {
+  /** @type {{email:string, token: string, password1: string, password2: string}} */
+  const data = req.body;
+
+  const passwordChanged = await userService.resetPassword(data.email, data.token, data.password1, data.password2);
+
+  res.send(passwordChanged);
+}));
+
+/**
+ * Send's to the user a password reset email.
+ */
+router.put("/auth/send-reset-password-email", apiAsyncWrapper(async (req, res) => {
+  /** @type {{email:string, passwordResetLinkPage: string}} */
+  const data = req.body;
+
+  const token = await userService.retrievePasswordResetToken(data.email);
+
+  if (token) {
+    await EmailService
+      .to(data.email)
+      .sendResetPasswordEmail(data.email, token, data.passwordResetLinkPage);
+
+    res.send(true);
+  } else {
+    res.send(false);
```

Comment:
We should not indicate whether or not the user exists in our database.

It should look like the password reset email was sent even if the user DOES NOT exist in our database.  This prevents the system from leaking information about what users are registered.